src/p2p/behaviour.rs

```diff
@@ -550,6 +556,72 @@ impl<Types: IpfsTypes> Behaviour<Types> {
             }
         }
     }
+
+    pub fn get_bootstrappers(&self) -> Vec<Multiaddr> {
+        self.swarm
+            .bootstrappers
+            .iter()
+            .cloned()
+            .map(|a| a.into())
+            .collect()
+    }
+
+    pub fn add_bootstrapper(
+        &mut self,
+        addr: MultiaddrWithPeerId,
+    ) -> Result<Multiaddr, anyhow::Error> {
+        let ret = addr.clone().into();
+        self.swarm.bootstrappers.insert(addr.clone());
+        let MultiaddrWithPeerId {
+            multiaddr: _,
+            peer_id,
+        } = addr.clone();
+        self.kademlia.add_address(&peer_id, addr.into());
+        Ok(ret)
+    }
+
+    pub fn remove_bootstrapper(
+        &mut self,
+        addr: MultiaddrWithPeerId,
+    ) -> Result<Multiaddr, anyhow::Error> {
+        let ret = addr.clone().into();
+        self.swarm.bootstrappers.remove(&addr);
+        Ok(ret)
+    }
+
+    pub fn clear_bootstrappers(&mut self) -> Vec<Multiaddr> {
+        self.swarm.bootstrappers.drain().map(|a| a.into()).collect()
+    }
+
+    pub fn restore_bootstrappers(&mut self) -> Vec<Multiaddr> {
+        let config_location = std::env::var("IPFS_PATH").expect("always expected to be set");
```

Comment:
there's an error in the path here, which is already fixed in the incoming follow-up; also, the `expect`s, however unlikely, are made into nicer errors